backend/src/controllers/orders-controller.ts

```diff
@@ -52,65 +52,120 @@ export const createOrder = async (req: CustomRequest, res: Response) => {
+        res.status(200).json({message: 'Order successfully canceled'});
     } catch (error) {
         console.error('Error canceling order:', error);
-        res.status(500).json({ error: 'Internal Server Error' });
+        res.status(500).json({error: 'Internal Server Error'});
     }
 };
-export const completeOrder = async (req: Request, res: Response) => {
+export const completeOrder = async (req: CustomRequest, res: Response) => {
     try {
         const orderId: string = req.params.id;
         const order: IOrder | null = await Order.findById(orderId);
         if (!order) {
-            res.status(404).json({ error: 'Order not found' });
+            res.status(404).json({error: 'Order not found'});
         }
         const seller = await User.findById(order?.itemOwnerId);
         if (!seller) {
-            return res.status(404).json({ error: 'Seller not found' });
+            return res.status(404).json({error: 'Seller not found'});
         }
-        await Order.updateOne({ _id: new ObjectId(orderId.trim()) }, { $set: { status: 'completed' } });
+        await Order.updateOne({_id: new ObjectId(orderId.trim())}, {$set: {status: 'completed'}});
         await sendEmail(order?.customerEmail, "Customer: Order Completed", "Your order ${orderId} has been completed.", `<p>Your order ${orderId} has been completed.</p>`)
         await sendEmail(seller.email, "Seller: Order Completed", "Order ${orderId} has been completed by ${order?.customerEmail}.", `<p>Order ${orderId} has been completed.</p>`)
-        res.status(200).json({ message: 'Order successfully completed' });
+        res.status(200).json({message: 'Order successfully completed'});
+    } catch (error) {
+        console.error('Error completing the order:', error);
+        res.status(500).json({error: 'Internal Server Error'});
+    }
+};
+export const updateOrder = async (req: CustomRequest, res: Response) => {
+    const {
+        newOrderNumber,
+        newCustomerName,
+        newCustomerEmail,
+        newCustomerType,
+        newSchool,
+        newNotes
+    }: {
+        newOrderNumber: number,
+        newCustomerName: string,
+        newCustomerEmail: string,
+        newCustomerType: string,
+        newSchool: string,
+        newNotes: string
+    } = req.body;
+    const orderId: string = req.params.id;
+    const user = await User.findOne(req.token?.userId);
+
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        return res.status(400).json({ errors: errors.array() });
+    }
+    if (!req.token?.userId || (user?.group !== 'buyer' && user?.group !== 'seller')) {
+        return res.status(403).json({ error: 'Invalid or missing user type' });
+    }
+    const order: IOrder | null = await Order.findById(orderId);
+    if (!order) {
+        return res.status(404).json({ error: 'Order not found' });
+    }
+    if (user?.group === 'buyer' && user?.group !== order.userWhoPlacedOrderId.toString()) {
```

Comment:
The second condition (user.group !== ...) will always fail, as `user.group` is always `buyer` or `seller`. 
`req.token.userId !== order.userWhoPlacedOrderId.toString()` on the second line is the correct way to implement this